src/linux/backend-v4l2.cpp

```diff
@@ -583,172 +576,260 @@ namespace librealsense
+            uint16_t vid{}, pid{}, mi{};
+            usb_spec usb_specification(usb_undefined);
+            std::string bus_info, card;
 
-                    uvc_device_info info{};
-                    info.pid = pid;
-                    info.vid = vid;
-                    info.mi = mi;
-                    info.id = dev_name;
-                    info.device_path = std::string(buff);
-                    info.unique_id = busnum + "-" + devpath + "-" + devnum;
-                    info.conn_spec = usb_specification;
-                    info.uvc_capabilities = get_dev_capabilities(dev_name);
+            auto dev_name = "/dev/" + name;
+
+            get_mipi_device_info(dev_name, bus_info, card);
 
-                    //std::cout << "Device " << name << ":\n" << std::string(info);
+            // D457-specific
+            // the follwing 2 lines need to be changed in order to enable multiple mipi devices support
+            // or maybe another field in the info structure - TBD
+            vid = 0x8086;
+            pid = 0xABCD; // D457 dev
 
+            static std::regex video_dev_index("\\d+$");
+            std::smatch match;
+            uint8_t ind{};
+            if (std::regex_search(name, match, video_dev_index))
+            {
+                ind = static_cast<uint8_t>(std::stoi(match[0]));
+            }
+            else
+            {
+                LOG_WARNING("Unresolved Video4Linux device pattern: " << name << ", device is skipped");
+                throw linux_backend_exception("Unresolved Video4Linux device, device is skipped");
+            }
+
+            //  D457 exposes (assuming first_video_index = 0):
+            // - video0 for Depth and video1 for Depth's md.
+            // - video2 for RGB and video3 for RGB's md.
+            // - video4 for IR stream. IR's md is currently not available.
+            // - video5 for IMU (accel or gyro TBD)
+            // next several lines permit to use D457 even if a usb device has already "taken" the video0,1,2 (for example)
+            // further development is needed to permit use of several mipi devices
+            static int  first_video_index = ind;
+            if (ind == first_video_index || ind == first_video_index + 2 || ind == first_video_index + 4)
+                mi = 0;
+            else if (ind == first_video_index + 1 | ind == first_video_index + 3)
+                mi = 3;
+            else if (ind == first_video_index + 5)
+                mi = 4;
+            else
+            {
+                LOG_WARNING("Unresolved Video4Linux device mi, device is skipped");
+                throw linux_backend_exception("Unresolved Video4Linux device, device is skipped");
+            }
+
+            uvc_device_info info{};
+            info.pid = pid;
+            info.vid = vid;
+            info.mi = mi;
+            info.id = dev_name;
+            info.device_path = video_path;
+            //info.unique_id = busnum + "-" + devpath + "-" + devnum;
```

Comment:
`unique_id` is not needed for mipi?
maybe add a comment explaining why is this commented out?

Reply:
The busnum, devpath and devnum used for this unique_id are available only via usb.
If we want to generate a unique id also for mipi, using other parameters, this can be done, but we need to think about it - I think this can be part of a further task - @Nir-Az do you agree?

Reply:
Agreed, but maybe we'll add a comment saying that?

Reply:
Correct